library/sensu_check_dict.py

```diff
@@ -44,7 +44,11 @@ def main():
                     module.params['name']: module.params['check']
                 }
             })
-
+            only_on_ip = check['checks'][module.params['name']].pop('only_on_ip', '')
+            if only_on_ip !='':
```

Comment:
```
only_on_ip = check['checks'][module.params['name']].pop('only_on_ip', '')
```
this already provide default vaule as '' when no key 'only_on_ip' is existed, so !='' works.   For valid IP validation, I can add it.